models/common.py

```diff
@@ -52,4 +54,72 @@ def forward(self, x):
         return self.block(x)
 
 
+class DeepLabHead(nn.Sequential):
+    def __init__(self, in_channels, num_classes):
+        super(DeepLabHead, self).__init__(
+            ASPP(in_channels, [12, 24, 36]),
+            nn.Conv2d(256, 256, 3, padding=1, bias=False),
+            nn.BatchNorm2d(256),
+            nn.ReLU(),
+            nn.Conv2d(256, num_classes, 1),
+            nn.Tanh()
```

Comment:
I'm curious as to why exactly is this new code is exactly the same as in the original [DeepLabV3 code](https://github.com/pytorch/vision/blob/master/torchvision/models/segmentation/deeplabv3.py), except for the `nn.Tanh()` line?

Reply:
Yes you are right very similar code. The reasoning behind it is FineTuning. Since the actual pretrained weights is being called. Some of the modules and layers of DeepLabV3 has to be modified for our use case.
1.) First modification is on the first layer to take an additional channel. An additional weight dimesion was initialized and appended to the exisiting conv filters effectively changing the first conv size from 64, 3, 7, 7 --> 64, 4, 7, 7
2.) Second modification is on the classifier layer which is why that piece of code exists. For the pretrained weights the classifier layer explicitily supports 21 classes (trained on Coco Set) whereby we are predictiing on 4 classes. So based on this piece of code we are overwriting the classifier layer to support any number of classes (4 classes in our own case). The idea is too retain the pretrained weights for all the other layers and modify first layer and last layers for our use case. Hence Finetuning for our use case.

Reply:
P.S. I would change the activation layer to Relu. Relu is much better for multi class problems